File: LeetCode/11.Container_With_Most_Water/11.Container_With_Most_Water.py
#!/usr/bin/env python3


class Solution:
    def maxArea(self, height: List[int]) -> int:
        max_water = 0
        i, j = 0, len(height) - 1
        while i < j:
            h = min(height[i], height[j])
            water = h * (j - i)
            max_water = max(max_water, water)
            if height[i] >= height[j]:
                j -= 1
            else:
                i += 1

        return max_water

File: LeetCode/11.Container_With_Most_Water/11.Container_With_Most_Water.swift
class Solution {
    func maxArea(_ height: [Int]) -> Int {
        // two pointer
        let n = height.count
        var l = 0, r = n - 1

        var res = 0
        while l < r {
            res = max(res, (r - l) * min(height[l], height[r]))

            if height[l] < height[r] {
                l += 1
            } else {
                r -= 1
            }
        }

        return res
    }
}
